### PR TITLE
File descriptor policies

### PR DESCRIPTION
**Pull request by rvansa**

### CODE REVIEW

**Comment by rvansa** [comment]
@AntonKozlov @jankratochvil Please review.

File: fd-policies.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+* `error`: The default action, just print error and fail the checkpoint.
+* `ignore`: Leave handling of the open file to C/R engine (CRIU). This will likely validate and reopen the file on restore.
+* `close`: Close the file. An attempt to use it after restore will fail with runtime exception.
+* `reopen`: Close the file, and try reopen it (on the same position) after restore.
+
+Unless the action is `error`, any file found open will trigger a warning to be printed to the logging system. This can be suppressed with `warn: false` property.
+
+## Pipes
+
+Anonymous pipes don't have any means to identify, therefore it makes sense to have at most one rule for these. Available actions are `error`, `ignore` and `close` with the same meaning as in case of files.
```
Comment (AntonKozlov) [inline fd-policies.md:41]: `ignore` likely will cause problems on restore. In what cases is that useful?

File: fd-policies.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# File descriptor policies
+
+CRaC requires that the application closes all open files, network connections etc. - on Linux these are represented as file descriptors. However, it might be difficult to alter the application to properly coordinate with the checkpoint, e.g. due to a code in a library you cannot modify. In those cases CRaC offers a limited handling via configuration. Note that this applies only to file descriptors opened through JDK API; anything opened through native code cannot be handled this way.
+
+The configuration is set up by pointing system property `jdk.crac.resource-policies` to a file that consists of several rules separated by three dashes (`---`). Lines starting with hash sign (`#`) are ignored. Each rule consists of several `key: value` pairs. The above is actually a subset of YAML format, so we suggest that you use the `.yaml` or `.yml` extension for convenient use in an editor. See an example of this file:
```
Comment (AntonKozlov) [inline fd-policies.md:5]: Need to mention the format, targets and actions are experimental and are subject to change.

File: fd-policies.md (hunk 1 above)
Comment (rvansa) [reply fd-policies.md:41]: The only application I can think of now is using `--inherit-fd` with that.

**Pull request merged**